### ISSUE TITLE
Discrepancy between vex.Open and vex.Parse

### ISSUE THREAD

**Issue by knqyf263** [openvex/go-vex#55]
`vex.Parse` assums the latest version.
https://github.com/openvex/go-vex/blob/eb8ba797f8bc61f5fe76f0ba37e1b3c51099b268/pkg/vex/functions_files.go#L33-L40

`vex.Open` autodetects the version.
https://github.com/openvex/go-vex/blob/eb8ba797f8bc61f5fe76f0ba37e1b3c51099b268/pkg/vex/functions_files.go#L96-L100

At first, I tried `vex.Open`, and my VEX file didn't work because the `@context` in the file was `"https://openvex.dev/ns/` even though the file content was compatible with v0.2.0. Then, `Open` fell back into v0.0.1.

But after I switched to `vex.Parse`, it suddenly worked. I was a bit surprised they behaved differently. Is it possible to change both to the same behavior? I'd like to know the reason if it is intentional. If you agree to the change, I can open a PR.

**Comment by puerco** [comment]
Yes this is the intended behavior, the [documentation](https://pkg.go.dev/github.com/openvex/go-vex/pkg/vex#Parse) could use some help but I think it explains it:

> func Parse(data []byte) (*VEX, error)
> Parse parses an OpenVEX document in the latest version from the data byte array.

> func Open(path string) (*VEX, error)
> Open tries to autodetect the vex format and open it

To try to understand better, here is the current status:

- Open is the universal opener for any openvex version or even other formats such as CSAF. 
- Load() and Parse() are equivalents for files and byte arrays respectively.


What you need is a `vex.Parse()` equivalent that understands any OpenVEX version?

**Comment by knqyf263** [comment]
I'm sorry for the late reply.

>Yes this is the intended behavior, the [documentation](https://pkg.go.dev/github.com/openvex/go-vex/pkg/vex#Parse) could use some help but I think it explains it:

Yes, I saw the comments, but I didn't understand why they differ. Loading from bytes or a file should be the same, no?

>What you need is a vex.Parse() equivalent that understands any OpenVEX version?

Right. For example, we're trying to consume VEX in OCI registries. Different OpenVEX versions may have been uploaded there. It is better to parse it in memory with `Parse()` rather than downloading the content to a local file and parsing the file with `Open()`.
